### DISCUSSION TITLE
Discord OAuth: tls failed to verify certificate

### DISCUSSION THREAD

**Discussion by jackskelt** [pocketbase/pocketbase#3584] [Q&A] [answered]
I'm using Pocketbase for Discord OAuth in the following way:

```ts
locals.pb.collection('users').authWithOAuth2Code("discord", "code", "codeVerifier", "redirect_uri");
```

In the Pocketbase instance, it is returning this:

```json
{
  "errorMessage": "Post \"https://discord.com/api/oauth2/token\": tls: failed to verify certificate: x509: certificate signed by unknown authority"
}
```

This is my `Dockerfile`. I tried hosting it on Fly.io and in a local Docker, and it results in the same error, but when I download and start it on my own OS, it works (Fedora OS 38).

```dockerfile
FROM alpine:latest as downloader

ARG TARGETOS
ARG TARGETARCH
ARG VERSION=0.19.0

ENV BUILDX_ARCH="${TARGETOS:-linux}_${TARGETARCH:-amd64}"

# Install the dependencies
RUN apk add --no-cache \
    unzip \
    ca-certificates \
    wget \
    zip \
    zlib-dev \
    openssh

RUN apk update && rm -rf /var/cache/apk/*

COPY 2835394.crt /usr/local/share/ca-certificates/2835394.crt

RUN chmod 644 /usr/local/share/ca-certificates/2835394.crt && update-ca-certificates

RUN wget https://github.com/pocketbase/pocketbase/releases/download/v${VERSION}/pocketbase_${VERSION}_${BUILDX_ARCH}.zip \
    && unzip pocketbase_${VERSION}_${BUILDX_ARCH}.zip \
    && chmod +x /pocketbase

FROM scratch

EXPOSE 8080

COPY --from=downloader /pocketbase /usr/local/bin/pocketbase
CMD ["/usr/local/bin/pocketbase", "serve", "--http=0.0.0.0:8080", "--dir=/pb_data", "--publicDir=/pb_public"]
```

Here, I'm adding a certificate that I downloaded from [crt.sh](https://crt.sh/?q=2835394). It doesn't work as well.

```dockerfile
COPY 2835394.crt /usr/local/share/ca-certificates/2835394.crt
```

I have already tried some things I found on the internet and in this repository, but nothing seems to work. What can I do?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure I follow. Why do you need to load a custom certificate manually if you use Fly.io?

The error is caused probably because the root certificates (aka. the `ca-certificates` package) seems to be installed only for the `downloader` image and not in the `scratch` build stage.
You can try to either copy the certificates or for simplicity use a single stage build in your Dockerfile similar to the one from https://github.com/pocketbase/pocketbase/discussions/537.

**Reply by jackskelt** [reply]
I didn't notice that. It works. Thank you so much!
